src/Controller/VocabularyController.php

```diff
@@ -117,7 +116,16 @@ public function add()
      * @param $lang string Language of the vocabulary items.
      */
     public function add_sentences($lang = null)
-    {
+    {   
+    
+        $filteredLang = $this->request->getSession()->read('filter_by_language');
+        if ($filteredLang == null && $lang != null) {
+            $this->request->getSession()->write('filter_by_language', $lang );
+        } else if ($filteredLang != null && $lang == null) {
+            $lang = $this->request->getSession()->read('filter_by_language');
+        } else if ($lang != null && $filteredLang != null && $lang != $filteredLang) {
+            $this->request->getSession()->write('filter_by_language', $lang);
+        }
```

Comment:
If you set the value of `$lang` here whereas the action parameter `$lang` was null, it means the page `/vocabulary/add_sentences` will show filtered languages. This is a problem because the URL says that it’s not filtered.

For example, let’s say I visit `/vocabulary/add_sentences` and I have filter_by_language=por stored in my session. It will show Portuguese vocabulary items to me. That’s good, but now let’s say I want to share this page to a friend. So I copy and paste the URL `/vocabulary/add_sentences` and send it to my friend. My friend opens the page, but the Portuguese vocabulary items won’t show up because my friend doesn’t have filter_by_language=por stored in session.

To solve this problem, you need to modify the link for "Sentences wanted" in `src/Template/Element/vocabulary/menu.ctp`, to make it point to `/vocabulary/add_sentences/por` if I have that value in session.